### PR TITLE
Fix/o21kb 37

### PR DESCRIPTION
**Pull request by shaynelar**
-can create edits to inventory items from the client
-styling is better
-created per field from validation, errors only show up on errored fields
-extracted error component into separate module
-page loads with the previous data for the inventory item

old pr was not letting me push to it anymore for some reason so check this instead (the other is closed)

### CODE REVIEW

File: client/src/pages/EditInventoryitem/EditInventoryItem.jsx [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+			warehouseName: { value: "Manhattan", error: null },
+		},
+		warehouseOptions: [],
+	};
+	
+	async componentDidMount() {
+		//TODO once single inventory item endpoint is done
+		const { id } = this.props.match.params;
+		try {
+			const response = await axios.get(`${API_URL}/inventory/${id}`);
```
Comment (nilouhou) [inline client/src/pages/EditInventoryitem/EditInventoryItem.jsx:31]: Later on, when the end point is ready, maybe you could use getData(url) from utils.js file

File: client/src/pages/EditInventoryitem/EditInventoryItem.jsx [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+										{this.state.fields.description.error && <FormError />}
+									</fieldset>
+									<fieldset className="edit-inventory-item__field">
+										<label
+											className="edit-inventory-item__label"
+											htmlFor="Category"
+										>
+											Category
+										</label>
+										<select className="edit-inventory-item__input edit-inventory-item__input--select">
```
Comment (nilouhou) [inline client/src/pages/EditInventoryitem/EditInventoryItem.jsx:188]: minor thing, select needs name tag,

**Comment by nilouhou** [comment]
Comments just for friendly reminder, but nothing to block for now, so I go ahead and merge the PR

File: client/src/pages/EditInventoryitem/EditInventoryItem.jsx (hunk 1 above)
Comment (shaynelar) [reply client/src/pages/EditInventoryitem/EditInventoryItem.jsx:31]: sure

**Pull request merged**